lib/elliptic/curve/mont.js

```diff
@@ -42,6 +45,12 @@ function Point(curve, x, z) {
 }
 inherits(Point, Base.BasePoint);
 
+MontCurve.prototype.decodePoint = function decodePoint(bytes, enc) {
+  bytes = utils.toArray(bytes, enc);
+
+  return this.point(bytes, 1);
```

Comment:
Well, I'd just take two branches of that `if` and put them into two methods. The preparation step might be put in some `_`-prefixed method in BasePoint.